benches/benches/set/blockchain.rs

```diff
@@ -32,6 +41,73 @@ use rand::{
     SeedableRng,
 };
 
+/// Reimplementation of `tempdir::TempDir` that allows creating a new
+/// instance without actually creating a new directory on the filesystem.
+/// This is needed since rocksdb requires empty directory for checkpoints.
+pub struct ShallowTempDir {
+    path: PathBuf,
+}
+impl ShallowTempDir {
+    pub fn new() -> Self {
+        let mut rng = rand::thread_rng();
+        let mut path = env::temp_dir();
+        path.push(format!("fuel-core-bench-rocksdb-{}", rng.next_u64()));
+        Self { path }
+    }
+}
+impl Drop for ShallowTempDir {
+    fn drop(&mut self) {
+        // Ignore errors
+        let _ = std::fs::remove_dir_all(&self.path);
+    }
+}
+
+pub struct BenchDb {
+    db: RocksDb,
+    /// Used for RAII cleanup. Contents of this directory are deleted on drop.
+    tmp_dir: ShallowTempDir,
+}
+
+impl BenchDb {
+    fn new(contract: &ContractId) -> io::Result<Self> {
+        let tmp_dir = ShallowTempDir::new();
+        let db = RocksDb::default_open(&tmp_dir.path, None).unwrap();
+        let db = Arc::new(db);
+        let mut database = Database::new(db.clone());
+        database.init_contract_state(
+            contract,
+            (0u64..10_000_000).map(|k| {
+                let mut key = Bytes32::zeroed();
+                key.as_mut()[..8].copy_from_slice(&k.to_be_bytes());
+                (key, key)
+            }),
+        )?;
+        drop(database);
+        Ok(Self {
+            tmp_dir,
+            db: Arc::into_inner(db).expect("All other references must be dropped"),
+        })
+    }
+
+    /// Create a new separate database instance using a rocksdb checkpoint
+    fn checkpoint(&self) -> Self {
```

Comment:
It seems you can combine this function with `into_vm_db` and return `VmDatabase` directly.

Reply:
We might need to have them separately if we do other setup functions, but I agree that YAGNI. Combined in 69bd9a52.